### PR TITLE
feat(app): validator set methods use consensus height

### PR DESCRIPTION
**Pull request by bastienfaivre**
Solves #70.

Refactored validator set storage to use height-based caching:

- **Added height parameter**: Both `get_validator_set()` and `set_validator_set()` now take a `Height` parameter to support querying validator sets for specific consensus heights
- **Height-related state**: Replaced `Option<ValidatorSet>` with `Option<(Height, ValidatorSet)>`.
- **Updated error handling**: `get_validator_set()` now returns `Option<&ValidatorSet>` with proper error handling at all call sites

This enables proper height-based validator set lookups.

PS: for now, the state only caches a single validator set for a given height.

### CODE REVIEW

File: app/src/state.rs [hunk 1]
```diff
@@ -262,7 +273,9 @@ impl State {
         let round = parts.round;
 
         // Get the expected proposer for this height and round
-        let validator_set = self.get_validator_set(); // TODO: Should pass height as a parameter
+        let validator_set = self
+            .get_validator_set(height)
+            .ok_or(ProposalValidationError::ValidatorSetNotFound { height })?;
```
Comment (jmalicevic) [inline app/src/state.rs]: Will we have a scenario where we are asked to use a validator set from the past ? I think at the moment this is not the case. So I am not sure whether we need this in the first place and if we do ,should we extend it to query here the smart contract to get the data.

That being said, maybe it is safer to have this error here in case there is a corner case so we can catch things.

Comment (jmalicevic) [reply app/src/state.rs]: Given the current logic I like this solution better. Please add a changelog entry and if the tests pass feel free to merge it.
